removeobsoletegpgkeys: Adjust for conversions

matejmatuska opened the pull request:
When doing upgrade + conversion, removing obsolete GPG from the current
(or target for that matter) distro doesn't make sense, because we are
moving to a different distro.
Instead, all distro provided keys from the source distro need to be
removed, as the target distro uses it's own keys. Those are imported
elsewhere later during the upgrade.

A new list is added to the gpg-signatures.json maps, which contains the
names of the fake RPMs "generated" upon importing a GPG key into the RPM
DB.
These are in the order the key IDs ("keys" in the map) are in, however
the mapping is not always 1:1 (e.g. the Centos SIG Extras keys).
Some are also missing, see TODO below.

The key id could be mapped to the RPM names, however since the RPM NVR
format is:
`gpg-pubkey-<last 8 chars from key ID>-<creation time of the signature packet>`
there could be a collision between the key IDs.

I addded annotations to the keys at:
https://github.com/oamg/leapp-repository/wiki/gpg%E2%80%90signatures.json-key-annotations.

Jira: RHEL-110190

matejmatuska committed:
Update rhel gpg-signatures map

Add auxilliary key 2, auxilliary key 3 and the old signing key to
"keys". These were handled in obsoleted-keys, but not used here.

Add beta key to keys obsoleted in rhel 8.

matejmatuska committed:
Update centos gpg-signatures map

Add SIG Extras key to "keys".

matejmatuska committed:
removeobsoletegpgkeys: Adjust for converting

When doing upgrade + conversion, removing obsolete GPG from the current
(or target for that matter) distro doesn't make sense, because we are
moving to a different distro.
Instead, all distro provided keys from the source distro need to be
removed, as the target distro uses it's own keys. Those are imported
elsewhere later during the upgrade.

A new list is added to the gpg-signatures.json maps, which contains the
names of the fake RPMs "generated" upon importing a GPG key into the RPM
DB.
These are in the order the key IDs ("keys" in the map) are in, however
the mapping is not always 1:1 (e.g. the Centos SIG Extras keys).

The key id could be mapped to the RPM names, however since the RPM NVR
format is:
gpg-pubkey-<last 8 chars from key ID>-<creation time of the signature packet>
there could be a collision between the key IDs.

Some key RPMs are missing in Alma Linux map as I couldn't find out what
keys some the fingerprints correspond to.

Jira: RHEL-110190

I addded annotations to the keys at:
https://github.com/oamg/leapp-repository/wiki/gpg%E2%80%90signatures.json-key-annotations.

matejmatuska commented on repos/system_upgrade/common/files/distro/centos/gpg-signatures.json:
```diff
@@ -1,8 +1,14 @@
 {
     "keys": [
         "24c6a8a7f4a80eb5",
+        "4eb84e71f2ee9d55",
```
I will probably still add these to key-rpms, but I have yet to find the files somewhere.

matejmatuska commented:
Hello @yuravk, in this PR I extended the RPM GPG keys config for all distros, including Alma Linux, with a list of names of the fake RPMs which are created when the keys are imported to RPM DB (gpg-pubkey-XXXXXX-XXXXXX) to complement the existing list of fingerprints.

However, I am not sure what keys some of the fingerprints for Alma Linux correspond to.
If you can point me to the GPG key files (e.g. in AL compose), I could include them in this PR.

Specifically I am interested in these:
- `51d6647ec21ad6ea`
- `429785e181b961a5`
- `d07bf2a08d50eb66`

yuravk commented:
> Hello @yuravk, in this PR I extended the RPM GPG keys config for all distros, including Alma Linux, with a list of names of the fake RPMs which are created when the keys are imported to RPM DB (gpg-pubkey-XXXXXX-XXXXXX) to complement the existing list of fingerprints.
> 
> However, I am not sure what keys some of the fingerprints for Alma Linux correspond to. If you can point me to the GPG key files (e.g. in AL compose), I could include them in this PR.
> 
> Specifically I am interested in these:
> 
> * `51d6647ec21ad6ea`
> * `429785e181b961a5`
> * `d07bf2a08d50eb66`

Hello @matejmatuska 

I believe they are:
* `429785e181b961a5` - https://repo.almalinux.org/elevate/RPM-GPG-KEY-ELevate (AlmaLinux ELevate key)
* `51d6647ec21ad6ea` - https://repo.almalinux.org/almalinux/RPM-GPG-KEY-AlmaLinux-legacy (AlmaLinux 8 old key)
* `d07bf2a08d50eb66` - https://repo.tuxcare.com/tuxcare/RPM-GPG-KEY-TuxCare (cloudlinux TuxCare key)

pirat89 reviewed (changes requested):
It looks good. there are just some stylistic problems `'` x `"`, one code simplification, and forgotten update of actor description. it looks good otherwise. I haven't tested it yet. Keeping it for the final rewview.

matejmatuska committed:
gpg/almalinux: Remove Eurolinux Tuxcare GPG key

Only RPM GPG keys provided by the distribution should be included in the
gpg-signatures.json files as they are used to detect which packages are
1st party.

The Eurolinux Tuxcare GPG key is 3rd party, therefore it should not be
included.

pirat89 reviewed (approved):
lgtm and seems to be working - tested manually for CS -> CS , CS -> RHEL (I forgot to make a snapshot of msgs from testing)
